IntegrationTests/Services/AWSDynamoDBIntegrationTests/AccountIDEndpointModeTests.swift

```diff
@@ -0,0 +1,149 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import AWSDynamoDB
+import SmithyTestUtil
+import SmithyIdentity
+import enum AWSClientRuntime.AccountIDEndpointMode
+import enum ClientRuntime.EndpointError
+
+final class AccountIDEndpointModeTests: XCTestCase {
```

Comment:
This is an integration test (although it doesn't actually hit the service.)

Account ID & endpoint mode are passed into the service client, while a dummy HTTP client (the same one used for protocol tests) is used to inspect the resulting HTTP request.